### ISSUE TITLE
density tests fail with "System error: The minimum allowed cpu-shares is 1024"

### ISSUE THREAD

**Issue by davidewatson** [kubernetes/kubernetes#25892]
I have a 10-node kubernetes 1.2 cluster running on AWS m3.medium instances with CoreOS 1010.1.0. When I run the kubernetes density e2e tests, containers fail to start with the error "System error: The minimum allowed cpu-shares is 1024". I have not seen this error when using AWS m3.large instances. According to the docker team, this issue is fixed in docker 1.11. Therefore we are asking CoreOS to back port the fix or support docker 1.11. Note that this failure is reliably reproducible.

A link to the ticket docker ticket:

https://github.com/docker/docker/issues/22818

An example of the error:

```
core@autoscaled ~ $ journalctl | grep cpu-shares
May 16 19:38:16 autoscaled dockerd[1083]: time="2016-05-16T19:38:16.992403355Z" level=error msg="Handler for POST /containers/f7cbef37f15fb830c5b71934ff3e410f81605bbd4a85124cf6d0aaaa95214008/start returned error: Cannot start container f7cbef37f15fb830c5b71934ff3e410f81605bbd4a85124cf6d0aaaa95214008: [9] System error: The minimum allowed cpu-shares is 1024"
May 16 19:38:16 autoscaled hyperkube[3827]: E0516 19:38:16.993365    3827 manager.go:1953] Error running pod "density30-bbaecf52-1b9d-11e6-978f-f45c898cf22b-es0fh_e2e-tests-density-mmkq0(bba4d015-1b9d-11e6-86f1-02580e2da74b)" container "density30-bbaecf52-1b9d-11e6-978f-f45c898cf22b": runContainer: API error (500): Cannot start container f7cbef37f15fb830c5b71934ff3e410f81605bbd4a85124cf6d0aaaa95214008: [9] System error: The minimum allowed cpu-shares is 1024
May 16 19:38:16 autoscaled hyperkube[3827]: E0516 19:38:16.993739    3827 pod_workers.go:138] Error syncing pod bba4d015-1b9d-11e6-86f1-02580e2da74b, skipping: failed to "StartContainer" for "density30-bbaecf52-1b9d-11e6-978f-f45c898cf22b" with RunContainerError: "runContainer: API error (500): Cannot start container f7cbef37f15fb830c5b71934ff3e410f81605bbd4a85124cf6d0aaaa95214008: [9] System error: The minimum allowed cpu-shares is 1024\n"
core@autoscaled ~ $
```

**Comment by vishh** [comment]
Why is the minimum `1024`? Is this systemd imposing this restriction?

**Comment by davidewatson** [comment]
Docker 1.10 changed the default cgroup driver (which we are using) to cgroupfs instead of systemd, so it is not a systemd bug. Also it doesn't not appear to always be enforced because some of the pods are scheduled with cpu.shares == 10. This is a link to the docker issue which fixes the problem in 1.11: https://github.com/opencontainers/runc/pull/491

**Comment by davidewatson** [comment]
This appears to be fixed with CoreOS version 1081.5.0 from the beta channel. Therefore I am closing...
